lib/sync/CoreIdCache.js

```diff
@@ -0,0 +1,66 @@
+/*
+* @module CoreIdCache
+*/
+export class CoreIdCache {
+  #sqlite
+  #tableName = 'coreIdCache'
+  #types = `'auth', 'data', 'blob'`
+  /**
+    * Create a CoreIdCache instance. This class handles a table to track cores belonging to different namespaces/stores (Authstore, Datastore, Blobstore)
+    * @param {import('../sqlite.js').Sqlite} sqlite an instance of better-sqlite3
+  */
+  constructor(sqlite){
+    this.#sqlite = sqlite
+    this.#sqlite.run(`
+    CREATE TABLE IF NOT EXISTS ${this.#tableName}
+                  (
+                    coreId TEXT PRIMARY KEY NOT NULL,
+                    namespace TEXT CHECK(namespace IN(${this.#types})) NOT NULL, 
+                    identityId TEXT NOT NULL
```

Comment:
Maybe index by discovery key (e.g. make discovery key the primary key). See https://github.com/digidem/mapeo-core-next/pull/55#issuecomment-1384001134 - we would likely want to lookup a core by discovery key.